### PR TITLE
[BH-1894] Create common ListViewWithLabels widget

### PR DESCRIPTION
**Pull request by dawidwojtas-mudita**
Move RelaxationListView to the common section.
Now the widget can be used by other applications.

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

File: products/BellHybrid/apps/application-bell-relaxation/ApplicationBellRelaxation.cpp [hunk 1]
```diff
@@ -86,19 +87,19 @@ namespace app
...
+                {"app_bell_relaxation_noises", paths::audio::proprietary() / paths::audio::colorOfNoises()},
+                {"app_bell_relaxation_uploaded_sounds", paths::audio::userApp() / paths::audio::relaxation()}};
             const auto pathsSortingVector = std::vector<SoundsRepository::PathSorting>{
                 {paths::audio::proprietary() / paths::audio::relaxation(), SoundsRepository::SortingBy::TitleAscending},
                 {paths::audio::proprietary() / paths::audio::colorOfNoises(),
                  SoundsRepository::SortingBy::TitleAscending},
                 {paths::audio::userApp() / paths::audio::relaxation(), SoundsRepository::SortingBy::TitleAscending}};
 
-            auto soundsRepository = std::make_unique<SoundsRepository>(app, pathsSortingVector);
+            auto soundsRepository = std::make_unique<SoundsRepository>(app, std::move(pathsSortingVector));
```
Comment (Maciej-Mudita) [inline products/BellHybrid/apps/application-bell-relaxation/ApplicationBellRelaxation.cpp:100]: `const auto` or `std::move`, never both

**Review by Maciej-Mudita — APPROVED** [review]

File: products/BellHybrid/apps/common/include/common/widgets/LabelOption.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include <common/widgets/LabelListItem.hpp>
+#include <common/widgets/ListViewWithLabels.hpp>
+#include <common/options/OptionBellMenu.hpp>
+namespace gui::option
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/include/common/widgets/LabelOption.hpp:9]: I'd add a newline between includes and beginning of the namespace as this seems to be a standard in our codebase

File: products/BellHybrid/apps/common/src/models/SongsModel.cpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    void SongsModel::updateRecordsCount()
+    {
+        songsRepository->updateFilesCount();
+    }
+
+    unsigned int SongsModel::requestRecordsCount()
+    {
+        return songsRepository->getFilesCount();
+    }
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/src/models/SongsModel.cpp:34]: Personally I prefer to separate each method with newline - that way it's IMO easier to read

File: products/BellHybrid/apps/common/src/models/SongsModel.cpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                                             nullptr};
+
+        return item.build();
+    }
+    void SongsModel::requestRecords(std::uint32_t offset, std::uint32_t limit)
+    {
+        songsRepository->getMusicFiles(
+            offset,
+            limit,
+            [this](const std::vector<db::multimedia_files::MultimediaFilesRecord> &records,
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/src/models/SongsModel.cpp:71]: IIRC you can use generic lambda here:
```suggestion
            [this](const auto &records, const auto repoRecordsCount)
```
that will somewhat simplify the notation, but that's rather a matter of personal preferences, so feel free to resolve it you prefer current form :wink:

File: products/BellHybrid/apps/common/src/widgets/LabelOption.cpp [hunk 5]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <common/widgets/LabelOption.hpp>
+
+#include <utility>
+namespace gui::option
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/src/widgets/LabelOption.cpp:7]: I'd add a newline between includes and beginning of the namespace as this seems to be a standard in our codebase

File: products/BellHybrid/apps/common/include/common/models/SongsModel.hpp [hunk 6]
```diff
@@ -3,44 +3,44 @@
...
     };
 
-    class RelaxationSongsModel : public RelaxationSongsProvider
+    class SongsModel : public SongsProvider
     {
       private:
         ApplicationCommon *application;
         std::unique_ptr<AbstractSoundsRepository> songsRepository;
-        std::map<MusicType, std::string> pathPrefixes;
+        std::map<std::string, std::string> pathPrefixes;
```
Comment (Lefucjusz) [inline products/BellHybrid/apps/common/include/common/models/SongsModel.hpp:30]: I'd think about adding some `using`s to typedef these strings to some more self-explanatory types - I'm not very familiar with this part of GUI and at a first glance it's quite hard to determine what this map contains when considering it from high-level (implementational) point of view

**Review by Lefucjusz — APPROVED** [review]

File: products/BellHybrid/apps/application-bell-relaxation/ApplicationBellRelaxation.cpp (hunk 1 above)
Comment (dawidwojtas-mudita) [reply products/BellHybrid/apps/application-bell-relaxation/ApplicationBellRelaxation.cpp]: Done! I left `const`

File: products/BellHybrid/apps/common/include/common/widgets/LabelOption.hpp (hunk 2 above)
Comment (dawidwojtas-mudita) [reply products/BellHybrid/apps/common/include/common/widgets/LabelOption.hpp:10]: Done

File: products/BellHybrid/apps/common/src/models/SongsModel.cpp (hunk 3 above)
Comment (dawidwojtas-mudita) [reply products/BellHybrid/apps/common/src/models/SongsModel.cpp:34]: Me too! I forgot, done ;)

File: products/BellHybrid/apps/common/src/models/SongsModel.cpp (hunk 4 above)
Comment (dawidwojtas-mudita) [reply products/BellHybrid/apps/common/src/models/SongsModel.cpp]: I simplified ;) Done

File: products/BellHybrid/apps/common/src/widgets/LabelOption.cpp (hunk 5 above)
Comment (dawidwojtas-mudita) [reply products/BellHybrid/apps/common/src/widgets/LabelOption.cpp:8]: Done

File: products/BellHybrid/apps/common/include/common/models/SongsModel.hpp (hunk 6 above)
Comment (dawidwojtas-mudita) [reply products/BellHybrid/apps/common/include/common/models/SongsModel.hpp]: I created an alias by `using` for this map. Done ;)

**Pull request merged**